c_glib/gandiva-glib/function-signature.cpp

```diff
@@ -0,0 +1,243 @@
+ *
+ * Returns: %TRUE if both of them have the same data, %FALSE otherwise.
+ *
+ * Since: 0.14.0
+ */
+gboolean
+ggandiva_function_signature_equal(GGandivaFunctionSignature *function_signature,
+                                  GGandivaFunctionSignature *other_function_signature)
+{
+  auto gandiva_function_signature =
+    ggandiva_function_signature_get_raw(function_signature);
+  auto gandiva_other_function_signature =
+    ggandiva_function_signature_get_raw(other_function_signature);
+
+  return (*gandiva_function_signature) == (*gandiva_other_function_signature);
+}
+
+/**
+ * ggandiva_function_signature_to_string:
+ * @function_signature: A #GGandivaFunctionSignature
+ *
+ * Returns: (transfer full): The string representation of the function signature.
+ *
+ *   It should be freed with g_free() when no longer needed.
+ *
+ * Since: 0.14.0
+ */
+gchar *
+ggandiva_function_signature_to_string(GGandivaFunctionSignature *function_signature)
+{
+  const auto gandiva_function_signature =
+    ggandiva_function_signature_get_raw(function_signature);
+  return g_strdup(gandiva_function_signature->ToString().c_str());
+}
+
+/**
+ * ggandiva_function_signature_get_return_type:
+ * @function_signature: A #GGandivaFunctionSignature
+ *
+ * Returns: (transfer full):
+ *   A #GArrowDataType of the return value of the function signature.
+ *
+ * Since: 0.14.0
+ */
+GArrowDataType *
+ggandiva_function_signature_get_return_type(GGandivaFunctionSignature *function_signature)
+{
+  const auto gandiva_function_signature =
+    ggandiva_function_signature_get_raw(function_signature);
+  auto arrow_data_type = gandiva_function_signature->ret_type();
+  auto data_type = garrow_data_type_new_raw(&arrow_data_type);
+  return data_type;
+}
+
+/**
+ * ggandiva_function_signature_get_base_name:
+ * @function_signature: A #GGandivaFunctionSignature
+ *
+ * Returns: (transfer full): A base name of the function signature.
+ *
+ *   It should be freed with g_free() when no longer needed.
+ *
+ * Since: 0.14.0
+ */
+gchar *
+ggandiva_function_signature_get_base_name(GGandivaFunctionSignature *function_signature)
+{
+  const auto gandiva_function_signature =
+    ggandiva_function_signature_get_raw(function_signature);
+  return g_strdup(gandiva_function_signature->base_name().c_str());
+}
+
+/**
+ * ggandiva_function_signature_get_param_types:
+ * @function_signature: A #GGandivaFunctionSignature
+ *
+ * Returns: (transfer full) (element-type GArrowDataType):
+ *   A list of parameter data types of the function signature.
+ *
+ * Since: 0.14.0
+ */
+GList *
+ggandiva_function_signature_get_param_types(GGandivaFunctionSignature *function_signature)
+{
+  const auto gandiva_function_signature =
+    ggandiva_function_signature_get_raw(function_signature);
+
+  GList *param_type_list = nullptr;
+  auto param_types = gandiva_function_signature->param_types();
+  for (auto i = param_types.begin(); i != param_types.end(); ++i) {
```

Comment:
Can we use `for (const auto &param_type : gandiva_function_signature->param_types())`?